### PR TITLE
Handling exceptions thrown on the server side

### PR DESCRIPTION
**Pull request by moozzyk**
Addresses: #62

### CODE REVIEW

File: src/Microsoft.AspNetCore.SignalR/Hub.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ public virtual Task OnConnectedAsync()
             return TaskCache.CompletedTask;
         }
 
-        public virtual Task OnDisconnectedAsync()
+        public virtual Task OnDisconnectedAsync(Exception exception)
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.SignalR/Hub.cs]: Hmmm, what would somebody do with this exception besides log it?

Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR/Hub.cs]: This is to let them know that the connection was not closed cleanly.

Comment (davidfowl) [inline src/Microsoft.AspNetCore.SignalR/Hub.cs]: Meh, I guess it's harmless.

**Comment by moozzyk** [comment]
Added a test.

**Pull request merged**